services/distribution/src/main/java/app/coronawarn/server/services/distribution/assembly/component/DccRevocationListStructureProvider.java

```diff
@@ -0,0 +1,192 @@
+import app.coronawarn.server.services.distribution.assembly.structure.archive.ArchiveOnDisk;
+import app.coronawarn.server.services.distribution.assembly.structure.archive.decorator.signing.DistributionArchiveSigningDecorator;
+import app.coronawarn.server.services.distribution.assembly.structure.directory.Directory;
+import app.coronawarn.server.services.distribution.assembly.structure.directory.DirectoryOnDisk;
+import app.coronawarn.server.services.distribution.assembly.structure.file.FileOnDisk;
+import app.coronawarn.server.services.distribution.config.DistributionServiceConfig;
+import app.coronawarn.server.services.distribution.dcc.DccRevocationClient;
+import app.coronawarn.server.services.distribution.dcc.DccRevocationListToProtobufMapping;
+import app.coronawarn.server.services.distribution.dcc.FetchDccListException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import org.bouncycastle.util.encoders.Hex;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+@Component
+public class DccRevocationListStructureProvider {
+
+  private static final Logger logger = LoggerFactory.getLogger(DccRevocationListStructureProvider.class);
+  private static final String KID_ARCHIVE = "kid";
+  private static final String CHUNK = "chunk";
+
+  private final CryptoProvider cryptoProvider;
+  private final DistributionServiceConfig distributionServiceConfig;
+  private final DccRevocationClient dccRevocationClient;
+  private final DccRevocationListService dccRevocationListService;
+  private final DccRevocationListToProtobufMapping dccRevocationToProtobufMapping;
+
+  /**
+   * Creates the CDN structure for DCC Revocation list.
+   */
+  public DccRevocationListStructureProvider(CryptoProvider cryptoProvider,
+      DistributionServiceConfig distributionServiceConfig, DccRevocationClient dccRevocationClient,
+      DccRevocationListService dccRevocationListService,
+      DccRevocationListToProtobufMapping dccRevocationToProtobufMapping) {
+    this.cryptoProvider = cryptoProvider;
+    this.distributionServiceConfig = distributionServiceConfig;
+    this.dccRevocationClient = dccRevocationClient;
+    this.dccRevocationListService = dccRevocationListService;
+    this.dccRevocationToProtobufMapping = dccRevocationToProtobufMapping;
+  }
+
+  /**
+   * Fetch DCC Revocation List.
+   */
+  public void fetchDccRevocationList() {
+    try {
+      Optional<List<RevocationEntry>> revocationEntryList = dccRevocationClient.getDccRevocationList();
+      revocationEntryList.ifPresent(revocationList -> dccRevocationListService.store(revocationList));
+    } catch (FetchDccListException e) {
+      logger.warn("Fetching DCC Revocation List failed. ", e);
+    } catch (Exception e) {
+      logger.warn("Storing DCC Revocation List failed. ", e);
```

Comment:
any reason, why this isn't logged as 'error'?

Reply:
no, we can use error, will update it